File: haiku/text.cpp
// 9 april 2015
#include <cstdlib>
#include <cstring>
#include "uipriv_haiku.hpp"
using namespace std;

char *uiHaikuStrdupText(const char *t)
{
	return strdup(t);
}

void uiFreeText(char *t)
{
	free(t);
}

File: darwin/text.m
// 10 april 2015
#import "uipriv_darwin.h"

char *uiDarwinNSStringToText(NSString *s)
{
	char *out;

	out = strdup([s UTF8String]);
	if (out == NULL) {
		fprintf(stderr, "memory exhausted in uiDarwinNSStringToText()\n");
		abort();
	}
	return out;
}

void uiFreeText(char *s)
{
	free(s);
}

int uiprivStricmp(const char *a, const char *b)
{
	return strcasecmp(a, b);
}
